Gem/Code/Source/Components/AWSMetricsSubmissionComponent.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#include "AWSMetricsSubmissionComponent.h"
+
+#include <Multiplayer/IMultiplayer.h>
+#include <AWSMetricsBus.h>
+
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace MultiplayerSample
+{
+    using namespace Multiplayer;
+
+    AZ_CVAR(bool, sv_submit_aws_metrics, false, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Whether to submit metrics to the AWS backend");
+
+    void AWSMetricsSubmissionComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (const auto serializationContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializationContext->Class<AWSMetricsSubmissionComponent, Component>()
+                ->Field("AWS Metrics Submission Interval Hint in Seconds", &AWSMetricsSubmissionComponent::m_AWSMetricsSubmissionIntervalInSecHint)
+                ->Version(1);
+
+            if (const auto editContext = serializationContext->GetEditContext())
+            {
+                editContext->Class<AWSMetricsSubmissionComponent>("AWSMetricsSubmissionComponent",
+                    "Handles metrics submission via the AWSMetrics Gem")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "MultiplayerSample")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(nullptr, &AWSMetricsSubmissionComponent::m_AWSMetricsSubmissionIntervalInSecHint, "AWS Metrics Submission Interval Hint in Seconds",
+                        "Hint for the AWS metrics submission interval. "

```

Comment:
NP: Looks like a trailing white space here

Reply:
I'm trying to break the message into two lines since it's too long. The complete string looks like below:

"Hint for the AWS metrics submission interval. This interval will apply to the metrics that need to be submitted regulary like player connection count."

The white space is for separating the two sentences :)